src/invidious/search/query.cr

```diff
@@ -48,10 +51,18 @@ module Invidious::Search
     )
       # Get the raw search query string (common to all search types). In
       # Regular search mode, also look for the `search_query` URL parameter
-      if @type.regular?
-        @raw_query = params["q"]? || params["search_query"]? || ""
-      else
-        @raw_query = params["q"]? || ""
+      _raw_query = params["q"]?
+      _raw_query ||= params["search_query"]? if @type.regular?
+      _raw_query ||= ""
+
+      # Remove surrounding whitespaces. Mostly useful for copy/pasted URLs.
+      @raw_query = _raw_query.strip
```

Comment:
Shouldn't this be under an `@inhibit_ssf` check as so we're just sending the query as-is when its off?

Reply:
if the query starts with a space, the check below becomes useless, so we need to remove at least the leading spaces.

Reply:
That's a good point, with the way as is though is sending a raw query like  `" youtube.com/watch?v=dQw4w9WgXcQ"` impossible then?

Reply:
Yes, though Youtube pretty much ignores the extra space. Search result for "query" and " query" are identical on average (they change from request to request, even with the exact same query)

Reply:
I realized that you can search for `\ youtube.com/xxx`, tho. It will disable smart search, then remove the backslash, and pass the rest of the string.